File: library/src/main/java/io/github/anderscheow/validator/rules/regex/DigitsRule.java
package io.github.anderscheow.validator.rules.regex;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import io.github.anderscheow.validator.rules.common.RegexRule;

public class DigitsRule extends RegexRule {

    private static final String DIGITS_REGEX = "\\d+";

    public DigitsRule() {
        super(DIGITS_REGEX, "Value does not match digits regex");
    }

    public DigitsRule(@StringRes int errorRes) {
        super(DIGITS_REGEX, errorRes);
    }

    public DigitsRule(@NonNull String errorMessage) {
        super(DIGITS_REGEX, errorMessage);
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/regex/DigitsRule.kt
package io.github.anderscheow.validator.rules.regex

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.common.RegexRule

class DigitsRule : RegexRule {

    constructor(@StringRes errorRes: Int) : super(DIGITS_REGEX, errorRes)

    constructor(errorMessage: String) : super(DIGITS_REGEX, errorMessage)

    companion object {
        private const val DIGITS_REGEX = "\\d+"
    }
}

fun digitsOnly(@StringRes errorRes: Int): DigitsRule = DigitsRule(errorRes)

fun digitsOnly(errorMessage: String): DigitsRule = DigitsRule(errorMessage)